analyses/chromosomal-instability/02b-plot-chr-instability-by-histology.Rmd

```diff
@@ -0,0 +1,195 @@
+
+### Usage
+
+This notebook can be run via the command line from the top directory of the 
+repository as follows:
+
+```
+Rscript -e "rmarkdown::render('analyses/chromosomal-instability/02b-plot-chr-instability-by-histology.Rmd', 
+                              clean = TRUE)"
+```
+
+### Set Up
+
+```{r}
+# Set seed so heatmaps turn out the same
+set.seed(2020)
+
+# Magrittr pipe
+`%>%` <- dplyr::`%>%`
+
+# Need the plotting functions
+source(file.path("util", "chr-break-plot.R"))
+```
+
+Set up option for how many samples minimum we need to include in CDF plot. 
+Borrowed this type of logic from @sjspielman in the [gene-set-enrichment analysis.](https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/ee1283991d30237a6a04be8fc6fec7672844133b/analyses/gene-set-enrichment-analysis/02-model-gsea.Rmd#L55)
+
+```{r}
+# Let's check if this is numeric
+if (!(is.numeric(params$min_samples))) {
+  warning("The parameter min_samples should be numeric. Trying to coerce to numeric.")
+  params$min_samples <- as.numeric(params$min_samples)
+} 
+```
+
+### Directories and Files
+
+```{r}
+# Path to data directory
+data_dir <- file.path("..", "..", "data")
+
+# Path to output directory
+plots_dir <- "plots"
+
+# Path to tumor type plots output directory
+hist_plots_dir <- file.path(plots_dir, "tumor-type")
+
+# Create the hist_plots_dir  if it does not exist
+if (!dir.exists(hist_plots_dir)) {
+  dir.create(hist_plots_dir, recursive = TRUE)
+}
+```
+
+Here's all the input files we will need: 
+
+```{r}
+metadata_file <- file.path(data_dir, "pbta-histologies.tsv")
+hist_binned_breaks_file <- file.path("breakpoint-data", "histology_breakpoint_binned_counts.RDS")
+total_densities_file <- file.path("breakpoint-data", "intersection_of_breaks_densities.tsv")
+```
+
+Output plot: 
+
+```{r}
+cdf_plot_file <- file.path(plots_dir, "breaks_cdf_plot.png")
+```
+
+### Read in data 
+
+Set up metadata
+
+```{r}
+# Read in the metadata
+metadata <- readr::read_tsv(metadata_file)
+```
+
+Load in the localized by histology breakpoint data. 
+
+```{r}
+hist_break_local <- readr::read_rds(hist_binned_breaks_file)
+```
+
+Read in the total intersection breaks densities TSV that was made by the 
+`00-setup-breakpoint-data.R` script. 
+
+```{r}
+total_break_densities_df <- readr::read_tsv(
+  total_densities_file
+) %>%
+  # Add on the short histology column from the metadata for plotting purposes
+  dplyr::inner_join(metadata %>% dplyr::select(Kids_First_Biospecimen_ID, short_histology),
+    by = c("samples" = "Kids_First_Biospecimen_ID")
+  )
+```
+
+## Plot CDF plot by histology
+
+```{r}
+breaks_cdf <- total_break_densities_df  %>%
+  # Only plot histologies groups with more than `min_samples` number of samples
+  dplyr::group_by(short_histology, add = TRUE) %>%
+  dplyr::filter(dplyr::n() > params$min_samples) %>%
+  # Calculate histology group mean
+  dplyr::mutate(
+    hist_mean = mean(breaks_density),
+    hist_rank = rank(breaks_density) / dplyr::n()
```

Comment:
Why is this divided by the group size? Oh... it is not the rank, but a relative rank, got it.